src/Strategy/Python/PyProjectGeneric.hs

```diff
@@ -0,0 +1,501 @@
+mkProject :: DiscoveredProjectType -> PyProjectProject -> DiscoveredProject PyProjectProject
+mkProject projectType project =
+  DiscoveredProject
+    { Types.projectType = projectType  -- Use the provided project type
+    , projectBuildTargets = mempty
+    , projectPath = projectDir project
+    , projectData = project
+    }
+
+-- | Parse a PyProject.toml file into a PyProjectGeneric
+parseGenericPyProject ::
+  ( Has ReadFS sig m
+  , Has Diagnostics sig m
+  ) =>
+  Path Abs File ->
+  m PyProjectGeneric
+parseGenericPyProject = readContentsToml
+
+-- | Analyze a PyProject project and extract dependencies
+-- Will use the lock file if available, otherwise falls back to pyproject.toml
+analyze ::
+  ( Has ReadFS sig m
+  , Has Diagnostics sig m
+  , Has Logger sig m
+  ) =>
+  Path Abs File ->
+  m DependencyResults
+analyze pyprojectFile = do
+  -- Parse the PyProject.toml file to get project metadata
+  let projectDir = parent pyprojectFile
+  pyproject <- parseGenericPyProject pyprojectFile
+  
+  -- Determine project type
+  let pyprojectType = Strategy.Python.PyProjectGeneric.Types.projectType pyproject
+  
+  -- Find potential lock files in the same directory
+  let pyprojectDir = parent pyprojectFile
+  -- Check for lock files in the directory
+  let pdmLockPath = pyprojectDir </> $(mkRelFile "pdm.lock")
+  let poetryLockPath = pyprojectDir </> $(mkRelFile "poetry.lock")
+  
+  pdmLockExists <- doesFileExist pdmLockPath
+  poetryLockExists <- doesFileExist poetryLockPath
+  
+  -- Create a lock files collection
+  let lockFiles = catMaybes
+        [ if pdmLockExists then Just (PDMLock, pdmLockPath) else Nothing
+        , if poetryLockExists then Just (PoetryLock, poetryLockPath) else Nothing
+        ]
+  
+  -- Create a temporary PyProjectProject to use with analyzeWithProject
+  let project = PyProjectProject
+        { pyprojectFile = pyprojectFile
+        , projectDir = projectDir
+        , lockFiles = lockFiles
+        , Strategy.Python.PyProjectGeneric.projectType = pyprojectType
+        }
+  
+  -- Use the common implementation
+  analyzeWithProject project
+
+-- | Extract dependencies from a PyProjectGeneric based on its project type
+extractDependencies :: PyProjectGeneric -> [Dependency]
+extractDependencies pyproject = 
+  case Strategy.Python.PyProjectGeneric.Types.projectType pyproject of
+    PoetryProject -> extractPoetryDependencies pyproject
+    PDMProject -> extractPDMDependencies pyproject
+    PEP621Project -> extractPEP621Dependencies pyproject
+    UnknownProject -> [] -- Unknown project type, no dependencies
+
+-- | Extract dependencies from a Poetry project using unified conversion functions
+extractPoetryDependencies :: PyProjectGeneric -> [Dependency]
+extractPoetryDependencies pyproject =
+  case poetrySection pyproject of
+    Nothing -> []
+    Just poetry -> 
+      let 
+        -- Convert production dependencies
```

Comment:
These comments don't really add anything over the names imo. Prefer to use comments to explain "why" and "how". Sometimes it's appropriate to describe "what" but that should be rare - ideally code is clear enough that "what" is easy.